### DISCUSSION TITLE
helmfile can't only get crd yaml

### DISCUSSION THREAD

**Discussion by zhaogaolong** [helmfile/helmfile#1824] [Questions] [answered]
helmfile doesn't have such a command for helm show crd, just gets the crd command.

Cause:
    We usually use the helmfile template to generate YAML, upload the YAML to the audit system, and then use `kubectl apply -f `to submit it. The crd is too large to be `kubectl apply`, and can only be submitted by `kubectl replate`, but other resources still want to be submitted to k8s by apply, so you need to get the yaml of the crd separately . Something like `helm show crds`

**Comment by mumoshu** [comment, marked as answer]
@zhaogaolong Hey!
I believe this is where combining `helmfile` with a tool like `yq` would be more effective than complicating `helmfile`.
What if you use this to grab only CRDs:

```
helmfile template | yq -e 'select(.kind == "CustomResourceDefinition")'
```

And this for everything else (notice the difference between `==` and `!=`)


```
helmfile template | yq -e 'select(.kind != "CustomResourceDefinition")'
```

You can pipe these into your audit system, and deploy via `kubectl replace -f -` and `kubectl apply -f -`  respectively.

**Reply by zhaogaolong** [reply]
nice 👍
